### ISSUE TITLE
Make a proper document in-tree out of the new jq language wiki

### ISSUE THREAD

**Issue by nicowilliams** [stedolan/jq#1801]
At the request of friends and colleagues, I've written a new description of the [jq programming language](https://github.com/stedolan/jq/wiki/jq-Language-Description) in a wiki.

We should bring this in-tree in the `docs/` directory, reformat as necessary, make it build, and publish it on the `gh_pages` site docs, including a link from the manual.

**Comment by nicowilliams** [comment]
@fadado Ah yes, I screwed up the `foreach` description.  I'll fix it later today, unless you do.  Thanks for the kind words, and any fixes you contribute.  If you have ideas for improvements, let me know!  Are there sections I should add?  Should we start copying in built-ins' docs from the manual into this one, with more organization?  (That last would be a ton of work...)

**Comment by nicowilliams** [comment]
@fadado I fixed the `foreach` thinko.

**Comment by fadado** [comment]
I agree with the following comment from the "jq Language Description":

    (XXX Perhaps we should remove all internals details?)

The whole section "Sub-expressions that are not Path Expressions" can be
moved to a better place.

JJOR
